### PR TITLE
boostification of proxy session

### PR DESCRIPTION
**Pull request by sima-fastly**
boostification of proxy session

tested manually

### CODE REVIEW

File: src/cpp/proxy/proxysession.h [hunk 1]
```diff
@@ -65,10 +66,9 @@ class ProxySession : public QObject
 	// takes ownership
 	void add(RequestSession *rs);
 
-signals:
-	void addNotAllowed(); // no more sharing, for whatever reason
-	void finished();
-	void requestSessionDestroyed(RequestSession *rs, bool accept);
+	Signal addNotAllowed; // no more sharing, for whatever reason
+	Signal finished;
+    boost::signals2::signal<void(RequestSession*, bool)> requestSessionDestroyed;
```
Comment (jkarneges) [inline src/cpp/proxy/proxysession.h:71]: Indentation

File: src/cpp/proxy/engine.cpp [hunk 2]
```diff
@@ -116,6 +116,9 @@ class Engine::Private : public QObject
 	Connection requestReadyConnection;
 	Connection socketReadyConnection;
 	Connection iRequestReadyConnection;
+	Connection addNotAllowedConnection;
+	Connection finishedConnection;
+	Connection reqSessionDestroyedConnection;
```
Comment (jkarneges) [inline src/cpp/proxy/engine.cpp:121]: There could be more than one of these.

File: src/cpp/proxy/engine.cpp [hunk 3]
```diff
@@ -776,13 +777,13 @@ private slots:
...
 
-	void ps_finished()
+	void ps_finished(ProxySession *ps)
 	{
-		ProxySession *ps = (ProxySession *)sender();
-
 		ProxyItem *i = proxyItemsBySession.value(ps);
 		assert(i);
 
+		finishedConnection.erase(ps);
```
Comment (jkarneges) [inline src/cpp/proxy/engine.cpp:785]: Need to delete from the other maps too.

Comment (sima-fastly) [reply src/cpp/proxy/engine.cpp:786]: thanks for catching this one.
I also simplified another conn map erase `d->finishedConnections.erase(req);`

**Review by jkarneges — APPROVED** [review]

**Pull request merged**